### DISCUSSION TITLE
Request for a clear Django 4.x.x support roadmap

### DISCUSSION THREAD

**Discussion by arneatec** [coddingtonbear/django-mailbox#272] [General]
With utmost respect for the maintainer and the contributors, I think it is in order to comment on the state of the project and hopefully provide a solution to move forward with django-mailbox.

Here are my two cents about the state of the project:
1. django-mailbox does not support latest Django (4.2.2 at the moment)
2. There are two issues that are showstoppers : 
    -   signal handling in signals.py - makes the app unusable unless local workaround is made 
    -   usage of the removed  force_text  (from django.utils.encoding import force_text) in the tests  - makes the tests unusable
3. After applying workarounds, the test pass ok with warnings:
```
Found 37 test(s).
System check identified no issues (0 silenced).
s.......Unknown encoding _iso-2022-jp$esc; interpreting as ASCII!
Unknown encoding _iso-2022-jp$esc encountered while decoding text payload.  Interpreting as ASCII with replacement, but some data may not be represented as the sender intended.
......Decoding error encountered; interpreting utf-8 as ASCII!
.Decoding error encountered; interpreting utf-8 as ASCII!
......Decoding error encountered; interpreting us-ascii as ASCII!
................
----------------------------------------------------------------------
Ran 37 tests in 0.209s
``` 
4. In my limited use case (Django 4.x.x with IMAP transport) django-mailbox appears to work fine. 

There are a lot of outstanding _pull_ _requests,_ some of which are superficial (dox, flake8, etc.), some are for new functionality and some are for the Django 4 compatibility.

The maintainer has repeatedly asked for **contributors to vet and test** the pull requests so the project can move forward. In my opinion, there needs to be a _clear separation of outstanding pull requests_ on one hand and the _Django compatibility_ on the other hand. 

Some of the end users (myself included) would like to have a (ideally fully) compatible django-mailbox for Django 4.x.x. and forward. This requires the showstoppers and other major issues related to compatibility to be resolved for 4.x.x and the solutions checked for regressions in previous versions.

I am sure that there will be contributors (meself included) that would participate in an effort like this.  

If there is a branch for Django 4 ongoing development and there is a clear roadmap present (probably authored by the maintainer and main contributors), then would-be-contributors can understand both the outstanding tasks and the road ahead. That may help with the current compatibility effort and further the confidence between users, contributors and maintainer. 
   
Therefore I suggest that a clear Django 4.x.x support roadmap be created and issues created off it so we can move forward.

**Comment by Pietro395** [comment]
I agree, I am also planning to migrate to Django 4 and I would have the opportunity to contribute and test.

@coddingtonbear Can we make a brach for django 4 by merging the pull requests that contribute to the support of the new version?

**Reply by coddingtonbear** [reply]
You do not need my permission for that; definitely go for it!

**Comment by coddingtonbear** [comment]
I'm afraid I don't have the time to allocate to this project at the moment, and it isn't likely to get close to the top of my priority list for the foreseeable future.  That said, if you and other folks, @arneatec & @skar395, put together some kind of roadmap, I'd be glad to give you collaborator access in exchange.

**Reply by Pietro395** [reply]
> I'm afraid I don't have the time to allocate to this project at the moment, and it isn't likely to get close to the top of my priority list for the foreseeable future. That said, if you and other folks, @arneatec & @skar395, put together some kind of roadmap, I'd be glad to give you collaborator access in exchange.

Unfortunately, I too have little time, but I will try in these days to evaluate the necessary changes
